test-suite/src/test/scala/org/scalajs/testsuite/javalib/LinkedHashSetTest.scala

```diff
@@ -0,0 +1,63 @@
+package org.scalajs.testsuite.javalib
+
+import java.{util => ju}
+
+import scala.collection.JavaConversions._
+
+class LinkedHashSetTest[F <: LinkedHashSetFactory](linkedHashSetFactory: F)
+  extends HashSetTest[F](linkedHashSetFactory) {
+
+  override def testApi(): Unit = {
+    super.testApi()
+
+    it("should iterate over elements in an ordered manner") {
+      val hs = new ju.LinkedHashSet[String]()
+
+      val l1 = List[String]("ONE", "TWO", (null: String))
+      expect(hs.addAll(asJavaCollection(l1))).toBeTruthy
+      expect(hs.size).toEqual(3)
+
+      val iter1 = hs.iterator()
+      val result1 = {
+        for {
+          i <- 0 until 3
+        } yield {
+          expect(iter1.hasNext()).toBeTruthy
+          val value = iter1.next()
+          expect(value).toEqual(l1(i))
+        }
```

Comment:
This `yield` yields the result of `.toEqual`, which is `()`. That is suspicious.